pkg/auth/auth.go

```diff
@@ -0,0 +1,173 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+
+	logging "github.com/op/go-logging"
+)
+
+// ConfigEntry - Configuration for authentication
+type ConfigEntry struct {
```

Comment:
I was conflicted about this because it is a list of auth definitions. 

```yaml
auth:
  - type: basic
    enabled: true
  - type: oauth
    enabled: true
  - type: cert
    enabled: true
```
So each type is a config entry. I guess we could say it's an array of auth.Config.